locality/models.py

```diff
@@ -0,0 +1,109 @@
+from __future__ import unicode_literals
+from django.db import models
+from django.utils.encoding import python_2_unicode_compatible
+
+
+@python_2_unicode_compatible
+class Locality(models.Model):
+    """
+    A base table that gives a globally unique ID to any
+    location (city, state, etc)
+    """
+    name = models.CharField(max_length=128)
+    short_name = models.CharField(max_length=32)
+
+    def __str__(self):
+        return self.name or self.short_name or str(self.id)
+
+    def dereference(self):
+        for rel in self._meta.get_all_related_objects():
+            if hasattr(self, rel.name):
+                return getattr(self, rel.name)
+        raise Exception("Abstract locality? %s" % self)
+
+    class Meta:
+        verbose_name_plural = 'localities'
+
+
+class FipsLocality(Locality):
+    """
+    Abstract class, for any model that has a fips_id
+    """
+    fips_id = models.IntegerField(null=True)
+
+    class Meta:
+        abstract = True
+
+
+class Precinct(Locality):
+    """
+    The smallest unit of geographic area for voters. Your precinct determines
+    who and what you vote on.
+    """
+    number = models.CharField(
+        max_length=5,
+        help_text="the precinct's number e.g., 32 or 32A "
+                  "(alpha characters are legal)."
+    )
+    zip_code = models.ForeignKey('ZipCode')
```

Comment:
I really don't know, but probably!

Broadening the topic: many of the models in `locality` aren't used. They were more for concept of what could be done, rather than being useful at the moment. Probably best to comment out, or remove, such models.